### PR TITLE
Windows nodeAllocatable test

### LINKED ISSUE #73417: Need System OOM testing on Windows
Kubelet and System-reserve are implemented on Windows along with pod/container reserves which can help prevent overprovisioning, but we still need to test OOM behavior. We need a test case that will:

1. Set kubelet & system reserve
2. Schedule pods until node is full
3. Run something that will take up system ram so that OOM is triggered

/sig windows
/milestone v1.14

### PR DESCRIPTION
**Pull request by lee0c**
Co-authored-by: Bruno Medina <bmedinab@gmail.com>

**What type of PR is this?**

/kind bug
/sig windows

**What this PR does / why we need it**:

This adds two tests that check on nodeAllocatable, systemReserve, hardEviction limits on windows agents. Checks that nodeAlloc is correct, then checks that it cannot deploy pods w/ mem limits past nodeAlloc re:

> I think the only valid test to be written for this scenario, as it stands now, is validating that Node Allocatable is being reported properly. [From issue discussion](https://github.com/kubernetes/kubernetes/issues/73417#issuecomment-465466288)

**Which issue(s) this PR fixes**:

Fixes #73417 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/windows/memory_limits.go [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+
+		pods = append(pods, &pod)
+	}
+
+	return pods
+}
+
+// getNodeMemory populates a nodeMemory struct with information from the first
+func getNodeMemory(f *framework.Framework) nodeMemory {
+	selector := labels.Set{"kubernetes.io/role": "agent"}.AsSelector()
```
Comment (benmoss) [inline test/e2e/windows/memory_limits.go]: I don't think it's standard or universal to have the role "agent" on nodes

Comment (lee0c) [reply test/e2e/windows/memory_limits.go]: Good to know, I'll switch the check to os=windows instead to align with the deployment

File: test/e2e/windows/memory_limits.go [hunk 2]
```diff
@@ -0,0 +1,336 @@
+/*
+Copyright 2018 The Kubernetes Authors.
```
Comment (neolit123) [inline test/e2e/windows/memory_limits.go]: 2019

**Comment by neolit123** [comment]
/ok-to-test
/priority important-soon

cc @jsturtevant
/assign @PatrickLang

File: test/e2e/windows/memory_limits.go (hunk 2 above)
Comment (lee0c) [reply test/e2e/windows/memory_limits.go]: Thanks, fixed it

**Review by michmike — APPROVED** [review]

**Comment by lee0c** [comment]
@michmike It says you approved this but it's missing the approved label - do I need someone else to approve it?

**Comment by michmike** [comment]
it needs patrick to approve as well. it will be soon

File: test/e2e/windows/memory_limits.go [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/client-go/kubernetes/scheme"
+	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
+	kubeletconfig "k8s.io/kubernetes/pkg/kubelet/apis/config"
+	"k8s.io/kubernetes/test/e2e/framework"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
```
Comment (danielqsj) [inline test/e2e/windows/memory_limits.go]: This cannot pass golint. Please fix it.

**Review by danielqsj — COMMENTED** [review]
/lgtm cancel
This is causing batches to fail `pull-kubernetes-verify` : https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/batch/pull-kubernetes-verify/124898
/cc spiffxp

File: test/e2e/windows/memory_limits.go (hunk 3 above)
Comment (lee0c) [reply test/e2e/windows/memory_limits.go]: Do you have a recommendation for what I should do to change it? Most of the tests I have been looking at use that same import so I'm unsure what is preferred

Comment (benmoss) [reply test/e2e/windows/memory_limits.go]: I think you have to use non-dot imports and refer to ginkgo/gomega methods with the package prefix.

Comment (lee0c) [reply test/e2e/windows/memory_limits.go]: Okay, wasn't sure if there were other options. I'll put that in in the next few hours

**Comment by neolit123** [comment]
> Using shellcheck 0.6.0 docker image.
Errors from shellcheck:
OCI runtime exec failed: exec failed: container_linux.go:348: starting container process caused "process_linux.go:90: adding pid 298615 to cgroups caused \"failed to write 298615 to cgroup.procs: write /sys/fs/cgroup/cpu,cpuacct/docker/9f41b95e98fd68c09611e4555e4f895eee691273cd364dac155a2f96090eb579/cgroup.procs: invalid argument\"": unknown

yikes.
/retest

**Comment by michmike** [comment]
thanks for fixing the issues quickly

**Pull request merged**